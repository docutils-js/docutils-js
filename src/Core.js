import readers from './Readers';
import writers from './Writers';
import { OptionParser } from './FrontEnd';
import { Source } from './Sources';
import { FileInput, FileOutput } from './io';
import Publisher from './Publisher';

export { Publisher };

export const defaultUsage = '%prog [options] [<source> [<destination>]]';
export const defaultDescription = ('Reads from <source> (default is stdin) and writes to <destination> (default is stdout).  See <http://docutils.sf.net/docs/user/config.html> for the full reference.');


export function publishCmdLine(args, cb) {
    const _defaults = {
 readerName: 'standalone',
                        parserName: 'restructuredtext',
                        usage: defaultUsage,
                        description: defaultDescription,
                        enableExitStatus: true,
};
    args = { ..._defaults, ...args };
    const {
 reader, readerName, parser, parserName, writer, writerName,
      settings, settingsSpec, settingsOverrides, configSection,
            enableExitStatus, argv, usage, description,
} = args;
    const pub = new Publisher({
 reader, parser, writer, settings,
});
    pub.setComponents(readerName, parserName, writerName);
    pub.publish({
argv, usage, description, settingsSpec, settingsOverrides, configSection, enableExitStatus,
}, cb);
}
