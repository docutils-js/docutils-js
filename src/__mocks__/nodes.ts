export const mockNoteCitation = jest.fn();
// @ts=ignore
const nodes = jest.requireActual('../nodes');

const mockDocument = jest.fn().mockImplementation(() => {
    return { noteCitation: mockNoteCitation };
});
nodes.document = mockDocument;

