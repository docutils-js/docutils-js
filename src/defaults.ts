const defaultReaderName = 'standalone';
const defaultParserName = 'restructuredtext';
const defaultEnableExitStatus = true;

export { defaultReaderName, defaultParserName, defaultEnableExitStatus };
