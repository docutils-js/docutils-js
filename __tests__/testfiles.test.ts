import { Publisher } from '../src/Core';
import { StringInput, StringOutput } from '../src/io';
import * as nodes from '../src/nodes';
import { defaults } from "../gen/defaults";
import {NodeInterface} from "../src/types";

const path = require('path');
const fs = require('fs');
const micromatch = require('micromatch');

const testFilesRoot = path.join(__dirname, '../testfiles/forms/');
//const glob = path.join(__dirname, '../testfiles/forms/*.txt');
// for files we want to return the files
// dor directories we want to return the directories

const table = fs.readdirSync(testFilesRoot, { withFileTypes: true }).filter((e: any) => !e.isDirectory()).map((e: any) => [e.name, fs.readFileSync(path.join(testFilesRoot, e.name), 'utf-8')]);

//const files = micromatch([glob]);
//const table = files.map(file => ([file, fs.readFileSync(file, 'utf-8')]));

const defaultArgs = {
    readerName: 'standalone',
    parserName: 'restructuredtext',
    usage: '',
    description: '',
    enableExitStatus: true,
    writerName: 'xml',
};

const defaultSettings = { ...defaults };

test.each(table)('%s', (file, input) => {
    const myOpts: any = {};

    const settings = { ...defaultSettings };
    const args = { ...defaultArgs };

    const { readerName, parserName, writerName } = args;
    const source = new StringInput( input);
    const destination = new StringOutput();
    const pub = new Publisher({
        source, destination, settings,
    });
    pub.setComponents(readerName, parserName, writerName);
    return new Promise((resolve, reject) => {
        /* {argv, usage, description, settingsSpec,
           settingsOverrides, configSection, enableExitStatus } */
        const fn = () => pub.publish({}, (error: any) => {
            if (error) {
                if (myOpts.expectError) {
                    resolve();
                } else {
                    reject(error);
                }
                return;
            }
            const document = pub.document;

            const Visitor = class extends nodes.GenericNodeVisitor {
                /* eslint-disable-next-line @typescript-eslint/camelcase,camelcase,@typescript-eslint/no-unused-vars,no-unused-vars */
                default_departure(node: NodeInterface) {
                    /**/
                }

                /* eslint-disable-next-line @typescript-eslint/camelcase,camelcase */
                default_visit(node: NodeInterface) {
                    if (node.attributes && node.attributes.refuri) {
                        //                                console.log(node.attributes.refuri);
                        if (!/^https?:\/\//.test(node.attributes.refuri)) {
                            const msg = `Invalid refuri ${node.attributes.refuri}`;
                            const messages = [document!.reporter.warning(msg, [], {})];
                            node.add(messages);
                        }
                    }
                }
            };
            const visitor = new Visitor(document!);
            document!.walkabout(visitor);
            expect(destination.destination).toMatchSnapshot();
            resolve();
        });
        fn();
    });
});
