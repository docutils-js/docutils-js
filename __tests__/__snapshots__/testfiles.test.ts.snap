// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`anonymous_reference.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<target ids=\\"1\\" refuri=\\"http://www.python.org\\" anonymous=\\"1\\"></target></document>"
`;

exports[`anonymous_via_two_dots.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<target ids=\\"1\\" refuri=\\"http://www.python.org\\" anonymous=\\"1\\"></target></document>"
`;

exports[`anonymous_via_twodots_no_blankfinish.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<target ids=\\"1\\" refuri=\\"http://www.python.org\\" anonymous=\\"1\\"></target></document>"
`;

exports[`block_quote.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<block_quote>
<paragraph>Block quotes consist of indented body elements:</paragraph><block_quote><paragraph>This theory, that is mine, is mine.</paragraph></block_quote><attribution>Anne Elk (Miss)</attribution></block_quote></document>"
`;

exports[`bullet_from_spec.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<bullet_list bullet=\\"-\\">
<list_item>
<paragraph>This is a bullet list.</paragraph></list_item><bullet_list bullet=\\"-\\"><list_item><paragraph>Bullets can be &quot;*&quot;, &quot;+&quot;, or &quot;-&quot;.</paragraph></list_item></bullet_list></bullet_list></document>"
`;

exports[`bullet_list__invalid_input.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<bullet_list bullet=\\"*\\">
<list_item>
<paragraph>bullet</paragraph></list_item><paragraph>invalid</paragraph></bullet_list><system_message line=\\"4\\"><paragraph>Bullet list ends without a blank line; unexpected unindent.</paragraph></system_message></document>"
`;

exports[`bullet_list__invalid_input_doctest.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<bullet_list bullet=\\"*\\">
<list_item>
<paragraph>bullet</paragraph></list_item><doctest_block>&gt;&gt;&gt; foo</doctest_block></bullet_list><system_message line=\\"4\\"><paragraph>Bullet list ends without a blank line; unexpected unindent.</paragraph></system_message></document>"
`;

exports[`bullet_list__invalid_input_field_marker.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<bullet_list bullet=\\"*\\">
<list_item>
<paragraph>bullet</paragraph></list_item><field_list><field><field_name>Hello</field_name><field_body><paragraph>foo</paragraph></field_body></field></field_list></bullet_list><system_message line=\\"5\\"><paragraph>Bullet list ends without a blank line; unexpected unindent.</paragraph></system_message></document>"
`;

exports[`bullet_list__invalid_input_line.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<bullet_list bullet=\\"*\\">
<list_item>
<paragraph>bullet</paragraph></list_item><transition></transition></bullet_list><system_message line=\\"4\\"><paragraph>Bullet list ends without a blank line; unexpected unindent.</paragraph></system_message></document>"
`;

exports[`bullet_no_unindent.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<bullet_list bullet=\\"*\\">
<list_item>
<paragraph>bullet</paragraph></list_item></bullet_list></document>"
`;

exports[`citation.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>Hello [Goober]_</paragraph><citation ids=\\"goober\\" names=\\"goober\\"><label>Goober</label><paragraph>Citation.</paragraph></citation></document>"
`;

exports[`comment.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<comment>Comments begin with two dots and a space.  Anything may
follow, except for the syntax of footnotes/citations,
hyperlink targets, directives, or substitution definitions.</comment></document>"
`;

exports[`complex.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<bullet_list bullet=\\"-\\">
<list_item>
<paragraph>This is the first line of a bullet list
item&apos;s paragraph.  All lines must align
relative to the first line.  [1]_</paragraph><block_quote><paragraph>This indented paragraph is interpreted
as a block quote.</paragraph></block_quote></list_item><paragraph>Because it is not sufficiently indented,
this paragraph does not belong to the list
item.</paragraph><footnote ids=\\"1\\" names=\\"1\\"><label>1</label><paragraph>Here&apos;s a footnote.  The second line is aligned
with the beginning of the footnote label.  The &quot;..&quot;
marker is what determines the indentation.</paragraph></footnote></bullet_list></document>"
`;

exports[`definition_list.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<definition_list>
<definition_list_item>
<term>what</term><definition><paragraph>Definition lists associate a term with a definition.</paragraph></definition></definition_list_item><definition_list><definition_list_item><term>how</term><definition><paragraph>The term is a one-line phrase, and the definition is one
or more paragraphs or body elements, indented relative to
the term.</paragraph></definition></definition_list_item></definition_list></definition_list></document>"
`;

exports[`definition_list_with_classifier_term.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<definition_list>
<definition_list_item>
<term>term</term><classifier>classifier</classifier><definition><paragraph>test</paragraph></definition></definition_list_item><definition_list><definition_list_item><term>what</term><definition><paragraph>Definition lists associate a term with a definition.</paragraph></definition></definition_list_item><definition_list><definition_list_item><term>how</term><definition><paragraph>The term is a one-line phrase, and the definition is one
or more paragraphs or body elements, indented relative to
the term.</paragraph></definition></definition_list_item></definition_list></definition_list></definition_list></document>"
`;

exports[`doctest_block.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<block_quote>
<doctest_block>&gt;&gt;&gt; print &apos;Python-specific usage examples; begun with &quot;&gt;&gt;&gt;&quot;&apos;
Python-specific usage examples; begun with &quot;&gt;&gt;&gt;&quot;
&gt;&gt;&gt; print &apos;(cut and pasted from interactive Python sessions)&apos;
(cut and pasted from interactive Python sessions)</doctest_block></block_quote></document>"
`;

exports[`docutils_title.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document ids=\\"docutil-project-documentation-overview\\" names=\\"docutils_\\\\ project\\\\ documentation\\\\ overview\\" source=\\"&lt;string&gt;\\" title=\\"Docutils_ Project Documentation Overview\\">
<title>Docutils_ Project Documentation Overview</title></document>"
`;

exports[`emphasis.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>*hello*</paragraph></document>"
`;

exports[`emphasis_and_inline.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>*hello* and **goodbye**</paragraph></document>"
`;

exports[`emphasis_followed_by_text.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>*hello* test</paragraph></document>"
`;

exports[`emphasis_preceded_by_text.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>stuff *hello*</paragraph></document>"
`;

exports[`emphasis_surrounded_by_text.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>stuff *hello* things</paragraph></document>"
`;

exports[`escaping_1.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>*escape* \`\`with\`\` &quot;&quot;</paragraph></document>"
`;

exports[`escaping_2.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>*escape* \`\`with\`\` &quot;\\\\&quot;</paragraph></document>"
`;

exports[`excerpt_1.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>Configuration settings:
\`footnote_references &lt;footnote_references setting_&gt;\`_.</paragraph><target ids=\\"footnote\\" names=\\"footnote\\" refuri=\\"../doctree.html#footnote\\"></target><target ids=\\"label\\" names=\\"label\\" refuri=\\"../doctree.html#label\\"></target><target ids=\\"footnote-reference-etting\\" names=\\"footnote_references\\\\ setting\\" refuri=\\"../../uer/config.html#footnote-reference-html4c1-writer\\"></target><system_message line=\\"11\\"><paragraph>Explicit markup ends without a blank line; unexpected unindent.</paragraph></system_message><system_message line=\\"12\\"><paragraph>Explicit markup ends without a blank line; unexpected unindent.</paragraph></system_message></document>"
`;

exports[`field_list.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<field_list>
<field>
<field_name>Author</field_name><field_body><paragraph>David Goodger</paragraph></field_body></field><field_list><field><field_name>Contact</field_name><field_body><paragraph>docutils-develop@lists.sourceforge.net</paragraph></field_body></field><field_list><field><field_name>Revision</field_name><field_body><paragraph>$Revision: 8205 $</paragraph></field_body></field><field_list><field><field_name>Date</field_name><field_body><paragraph>$Date: 2017-11-27 03:07:28 -0800 (Mon, 27 Nov 2017) $</paragraph></field_body></field><field_list><field><field_name>Copyright</field_name><field_body><paragraph>This document has been placed in the public domain.</paragraph></field_body></field></field_list></field_list><system_message line=\\"8\\"><paragraph>Field list ends without a blank line; unexpected unindent.</paragraph></system_message></field_list><system_message line=\\"9\\"><paragraph>Field list ends without a blank line; unexpected unindent.</paragraph></system_message></field_list><system_message line=\\"10\\"><paragraph>Field list ends without a blank line; unexpected unindent.</paragraph></system_message></field_list><system_message line=\\"11\\"><paragraph>Field list ends without a blank line; unexpected unindent.</paragraph></system_message></document>"
`;

exports[`footnote.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<footnote ids=\\"1\\" names=\\"1\\">
<label>1</label><paragraph>A footnote contains body elements, consistently
indented by at least 3 spaces.</paragraph></footnote></document>"
`;

exports[`footnote_auto-symbol.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>Here is a symbolic footnote reference: [*]_.</paragraph><footnote ids=\\"1\\" auto=\\"*\\"><paragraph>This is the footnote.</paragraph></footnote></document>"
`;

exports[`footnote_reference_strangeness.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>If [#note]_ is the first footnote reference, it will show up as
&quot;[1]&quot;.  We can refer to it again as [#note]_ and again see
&quot;[1]&quot;.  We can also refer to it as note_ (an ordinary internal
hyperlink reference).</paragraph><footnote ids=\\"note\\" names=\\"note\\" auto=\\"1\\"><paragraph>This is the footnote labeled &quot;note&quot;.</paragraph></footnote></document>"
`;

exports[`footnotes_multiple_refs.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>[#]_ is a reference to footnote 1, and [#]_ is a reference to
footnote 2.</paragraph><footnote ids=\\"1\\" auto=\\"1\\"><paragraph>This is footnote 1.</paragraph></footnote><footnote ids=\\"2\\" auto=\\"1\\"><paragraph>This is footnote 2.</paragraph></footnote><footnote ids=\\"3\\" auto=\\"1\\"><paragraph>This is footnote 3.</paragraph></footnote><paragraph>[#]_ is a reference to footnote 3.</paragraph><system_message line=\\"11\\"><paragraph>Explicit markup ends without a blank line; unexpected unindent.</paragraph></system_message><system_message line=\\"12\\"><paragraph>Explicit markup ends without a blank line; unexpected unindent.</paragraph></system_message></document>"
`;

exports[`image_directive.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<system_message line=\\"1\\">
<paragraph>Using English fallback for directive image</paragraph></system_message><system_message line=\\"1\\"><paragraph>Unknown directive type &quot;image&quot;.</paragraph><literal_block>.. image:: mylogo.png</literal_block></system_message></document>"
`;

exports[`incomplete_substitution_definition_with_image.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<comment>|beep</comment><system_message line=\\"1\\"><paragraph>malformed substitution definition.</paragraph></system_message></document>"
`;

exports[`inline_followed_by_emphasis.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>**hello** and *goodbye*</paragraph></document>"
`;

exports[`inline_internal_targets.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<bullet_list bullet=\\"*\\">
<list_item>
<paragraph>bullet list</paragraph><target ids=\\"econd-item-\\" names=\\"second\\\\ item\`\\"></target></list_item><bullet_list bullet=\\"*\\"><list_item><paragraph>second item, with hyperlink target.</paragraph></list_item></bullet_list></bullet_list></document>"
`;

exports[`interpreted_text_role_emphasis_prefix.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>:emphasis:\`test\`</paragraph></document>"
`;

exports[`interpreted_text_role_emphasis_suffix.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>\`test\`:emphasis:</paragraph></document>"
`;

exports[`interpreted_text_role_prefix.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>:role:\`test\`</paragraph></document>"
`;

exports[`interpreted_text_role_suffix.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>\`test\`:role:</paragraph></document>"
`;

exports[`interpreted_text_with_both_roles.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>:role:\`test\`:role:</paragraph></document>"
`;

exports[`interpreted_text_with_no_end_string_suffix.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>\`test</paragraph></document>"
`;

exports[`interpreted_text_with_no_role.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>\`test\`</paragraph></document>"
`;

exports[`line_block_with_continuation_line.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<line_block>
<line indent=\\"0\\">Lend us a couple of bob till Thursday.</line><line_block><line indent=\\"0\\">I&apos;m absolutely skint.</line><line_block><line indent=\\"0\\">But I&apos;m expecting a postal order and I can pay you back
as soon as it comes.</line><line_block><line indent=\\"0\\">Love, Ewan.</line></line_block></line_block><system_message line=\\"4\\"><paragraph>Line block ends without a blank line.</paragraph></system_message></line_block><system_message line=\\"3\\"><paragraph>Line block ends without a blank line.</paragraph></system_message></line_block><system_message line=\\"2\\"><paragraph>Line block ends without a blank line.</paragraph></system_message></document>"
`;

exports[`links.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<target ids=\\"a-re-tructuredtext-primer\\" names=\\"a\\\\ restructuredtext\\\\ primer\\" refuri=\\"../../uer/rt/quicktart.html\\"></target><target ids=\\"quick-re-tructuredtext\\" names=\\"quick\\\\ restructuredtext\\" refuri=\\"../../uer/rt/quickref.html\\"></target><system_message line=\\"5\\"><paragraph>Explicit markup ends without a blank line; unexpected unindent.</paragraph></system_message></document>"
`;

exports[`literal_block.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<block_quote>
<paragraph>Literal blocks are either indented or line-prefix-quoted blocks,
and indicated with a double-colon (&quot;::&quot;) at the end of the
preceding paragraph (right here --&gt;):</paragraph><literal_block>if literal_block:
    text = &apos;is left as-is&apos;
    spaces_and_linebreaks = &apos;are preserved&apos;
    markup_processing = None</literal_block></block_quote></document>"
`;

exports[`literal_block_without_blank_finish.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<block_quote>
<paragraph>Literal blocks are either indented or line-prefix-quoted blocks,
and indicated with a double-colon (&quot;::&quot;) at the end of the
preceding paragraph (right here --&gt;):</paragraph><literal_block>if literal_block:
    text = &apos;is left as-is&apos;
    spaces_and_linebreaks = &apos;are preserved&apos;
    markup_processing = None</literal_block></block_quote></document>"
`;

exports[`mixed_bullets.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<bullet_list bullet=\\"*\\">
<list_item>
<paragraph>bullet</paragraph></list_item><bullet_list bullet=\\"+\\"><list_item><paragraph>bullet</paragraph></list_item></bullet_list></bullet_list><system_message line=\\"5\\"><paragraph>Bullet list ends without a blank line; unexpected unindent.</paragraph></system_message></document>"
`;

exports[`multilevel_blockquote.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>This is a top-level paragraph.</paragraph><block_quote><paragraph>This paragraph belongs to a first-level block quote.</paragraph><block_quote><paragraph>This paragraph belongs to a second-level block quote.</paragraph></block_quote></block_quote><paragraph>Another top-level paragraph.</paragraph><block_quote><block_quote><paragraph>This paragraph belongs to a second-level block quote.</paragraph></block_quote><paragraph>This paragraph belongs to a first-level block quote.  The
second-level block quote above is inside this first-level
block quote.</paragraph></block_quote></document>"
`;

exports[`nested_bullets.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<bullet_list bullet=\\"*\\">
<list_item>
<paragraph>bullet</paragraph></list_item><block_quote><bullet_list bullet=\\"+\\"><list_item><paragraph>bullet</paragraph></list_item><bullet_list bullet=\\"+\\"><list_item><paragraph>bullet</paragraph></list_item></bullet_list></bullet_list></block_quote><bullet_list bullet=\\"*\\"><list_item><paragraph>bullet</paragraph></list_item></bullet_list></bullet_list></document>"
`;

exports[`nested_sections.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<section ids=\\"title\\" names=\\"title\\">
<title>Title</title><paragraph>Section-------</paragraph><paragraph>Third@@@@@</paragraph></section><section ids=\\"econd\\" names=\\"second\\"><title>Second</title><subtitle ids=\\"ooop\\" names=\\"ooops\\">Ooops</subtitle></section></document>"
`;

exports[`option_list.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<block_quote>
<option_list>
<option_list_item>
<option_group>
<option>
<option_string>-a</option_string></option></option_group><description><paragraph>command-line option &quot;a&quot;</paragraph></description></option_list_item><option_list><option_list_item><option_group><option><option_string>-b</option_string><option_argument delimiter=\\"\\"> file</option_argument></option></option_group><description><paragraph>options can have arguments
and long descriptions</paragraph></description></option_list_item><option_list><option_list_item><option_group><option><option_string>--long</option_string></option></option_group><description><paragraph>options can be long also</paragraph></description></option_list_item><option_list><option_list_item><option_group><option><option_string>--input</option_string><option_argument delimiter=\\"=\\">file</option_argument></option></option_group><description><paragraph>long options can also have
arguments</paragraph></description></option_list_item><option_list><option_list_item><option_group><option><option_string>/V</option_string></option></option_group><description><paragraph>DOS/VMS-style options too</paragraph></description></option_list_item></option_list></option_list><system_message line=\\"10\\"><paragraph>Option list ends without a blank line; unexpected unindent.</paragraph></system_message></option_list><system_message line=\\"11\\"><paragraph>Option list ends without a blank line; unexpected unindent.</paragraph></system_message></option_list><system_message line=\\"12\\"><paragraph>Option list ends without a blank line; unexpected unindent.</paragraph></system_message></option_list><system_message line=\\"13\\"><paragraph>Option list ends without a blank line; unexpected unindent.</paragraph></system_message></block_quote></document>"
`;

exports[`paragraph_ending_in___.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>This is my paragraph ending in:</paragraph><system_message line=\\"1\\"><paragraph>Literal block expected; none found.</paragraph></system_message></document>"
`;

exports[`program_lang.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>Want to learn about \`my favorite programming language\`_?</paragraph><target ids=\\"my-favorite-programming-language\\" names=\\"my\\\\ favorite\\\\ programming\\\\ language\\" refuri=\\"http://www.python.org\\"></target></document>"
`;

exports[`random.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<bullet_list bullet=\\"*\\">
<list_item>
<paragraph>bullet</paragraph></list_item><bullet_list bullet=\\"*\\"><list_item><paragraph>bullet</paragraph></list_item></bullet_list></bullet_list><system_message line=\\"7\\"><paragraph>Bullet list ends without a blank line; unexpected unindent.</paragraph></system_message></document>"
`;

exports[`random_2.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>Test.
Test2
Test3
-----</paragraph></document>"
`;

exports[`random_4.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document ids=\\"te-t3\\" names=\\"test3\\" source=\\"&lt;string&gt;\\" title=\\"Test3\\">
<title>Test3</title><paragraph>This is a test.</paragraph><bullet_list bullet=\\"*\\"><list_item><paragraph>BUllet list 1</paragraph></list_item><bullet_list bullet=\\"*\\"><list_item><paragraph>The emacs rst editor is weird.</paragraph></list_item></bullet_list></bullet_list><system_message line=\\"10\\"><paragraph>Bullet list ends without a blank line; unexpected unindent.</paragraph></system_message></document>"
`;

exports[`random_indent.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
</document>"
`;

exports[`short_overline.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<system_message line=\\"1\\">
<paragraph>Possible incomplete section title.
Treating the overline as ordinary text because it&apos;s so short.</paragraph></system_message><paragraph>===
Title
===</paragraph></document>"
`;

exports[`short_overline_2.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<system_message line=\\"1\\">
<paragraph>Possible incomplete section title.
Treating the overline as ordinary text because it&apos;s so short.</paragraph></system_message><paragraph>===
Title</paragraph></document>"
`;

exports[`simple_table.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<block_quote>
<table>
<tgroup cols=\\"3\\">
<colspec colwidth=\\"20\\">
</colspec><colspec colwidth=\\"10\\">
</colspec><colspec colwidth=\\"10\\">
</colspec><thead>
<row>
<entry>
<paragraph>Header row, column 1</paragraph></entry></row></thead><tbody><row><entry><paragraph>body row 1, column 1</paragraph></entry></row><row><entry><paragraph>body row 2</paragraph></entry></row></tbody></tgroup></table></block_quote></document>"
`;

exports[`simple_table_from_policies.txt.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<block_quote>
<table>
<tgroup cols=\\"3\\">
<colspec colwidth=\\"13\\">
</colspec><colspec colwidth=\\"7\\">
</colspec><colspec colwidth=\\"44\\">
</colspec><thead>
<row>
<entry>
<paragraph>Release Level</paragraph></entry></row></thead><tbody><row><entry><paragraph>alpha</paragraph></entry></row><row><entry><paragraph>beta</paragraph></entry></row><row><entry><paragraph>candidate</paragraph></entry></row><row><entry><paragraph>final</paragraph></entry></row></tbody></tgroup></table></block_quote></document>"
`;

exports[`strong.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>**hello**</paragraph></document>"
`;

exports[`substitution_definition_with_image.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>symbol here| image:: symbol.png</paragraph><system_message source=\\"\\" line=\\"1\\"><paragraph>Substitution definition &quot;symbol here&quot; empty or invalid.</paragraph><literal_block>.. |symbol here| image:: symbol.png</literal_block></system_message></document>"
`;

exports[`tabs.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>hello   tabs</paragraph></document>"
`;

exports[`title.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document ids=\\"title\\" names=\\"title\\" source=\\"&lt;string&gt;\\" title=\\"Title\\">
<title>Title</title><paragraph>Paragraph.</paragraph></document>"
`;

exports[`transition_correction.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>====:</paragraph><literal_block>====::</literal_block></document>"
`;

exports[`transition_marker.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<transition>
</transition></document>"
`;
